pkg/kubelet/kubelet.go

```diff
@@ -2438,6 +2438,15 @@ func (kl *Kubelet) fastStatusUpdateOnce() {
 			klog.ErrorS(err, "Error getting node")
 			continue
 		}
+		for _, c := range node.Status.Conditions {
+			if c.Type == v1.NodeReady {
+				if c.Status == v1.ConditionTrue {
+					return
+				} else {
+					break
```

Comment:
the side effect here is that if Node never becomes ready, we never leave this function. This may be a problem on a "broken" nodes where something will never come up. To be safe I'd suggest to time bound this loop.

Reply:
it is true that if PodCIDR is never successfully updated, it will have the same effect. But I think the PodCIDR is more contained issue than some potential issue we don't know about.

Reply:
What time bound value would you suggest? It can be also a log cutoff value I think, to resolve the logging issue. Maybe:

1. Add a time bound value. This function exits when the node is ready, or the time limit is reached, whichever earlier.
2. In updateRuntimeUp, log the confusing error as a log at higher verbosity while fastStatusUpdateOnce, or an error after fastStatusUpdateOnce has exited (assuming the time limit is a reasonable point that the node should be ready by - or the node was ready at one point and becomes not ready at a later point).

Also the todo is now done.

Reply:
The logic LGTM.

Can we use `wait.PollImmediate` or wait.* instead of for loop here? 
- interval: 100 * time.Millisecond
- timeout: 120 * time.Second

Reply:
Using wait.* looks like a good idea. I was just following the previous pattern of the code there.

containerRuntimeReadyExpected is probably still needed, because the verbose log is made inside updateRuntimeUp, AND this function is also called by someone else: https://github.com/kubernetes/kubernetes/blob/98233be715f78c22c24caf1158e51c4beb1f8644/pkg/kubelet/kubelet.go#L1450 before the wait times out here. We also want to prevent updateRuntimeUp called by others printing the verbose log before nodeReadyGracePeriod ends.

Reply:
Had another look, and I found it not as good as I thought to use wait.*. Specifically I'm talking about: "Some intervals may be missed if the condition takes too long or the time window is too short."

This means if status update takes a bit longer (taking the most of 100ms), this function would almost running nonstop. This would put an even heavier load on other parts of kubelet or the system.

Is there a similar function that would guarantee some gap between each poll?

Reply:
> But I think the PodCIDR is more contained issue than some potential issue we don't know about.

commenting without full context, so forgive me if is unrelated, but you are not guaranteed to have the PodCIDR set, is not required in all environments

Reply:
I see, so this is an improvement, 👍

Reply:
Moving
```
		kl.updateRuntimeUp()
		kl.syncNodeStatus()
```
out of the `if` given "you are not guaranteed to have the PodCIDR set" mentioned above.